docs/design/2024-10-11-put-and-verify-transactions-for-external-storages.md

```diff
@@ -0,0 +1,233 @@
+# Put and Verify Transactions for External Storages
+
+- Author: [Yu Juncen](https://github.com/YuJuncen)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/56523
+
+## Background 
+
+Sometimes, we need to control concurrency access to the same backup archive, like:
+
+- When compacting / restoring, we want to block migrating to a new version. 
+- When migrating the backup storage to a new version, we want to forbid reading.
+- When truncating the storage, we don't want another trancating operation happen.
+- When backing up, we don't want another backup uses the same storage.
+
+But external storage locking isn't trivial. Simply putting a lock file isn't safe enough: because after checking there isn't such a lock file, another one may write it immediately. Object locks provide stronger consistency, but also require extra configuration and permissions.
```

Comment:
hmm, I thought the new feature `conditional put` from S3 can solve this race condition, other vendors also have it. I remember you mentioned this feature can provide RW support, maybe that's where it differs from simply using `conditional put` on a lock file? I feel like it could be better to mention the comparison between the two.